### PR TITLE
rpcsrv: add SessionEnabled, MaxIteratorResultItems to `getversion`

### LINKED ISSUE #3276: Extend `getversion` RPC response with RPC server settings
Ref. https://github.com/nspcc-dev/neo-go/pull/3274#issuecomment-1864716933. We need to add a set of specific RPC server settings to `getversion` RPC response: `SessionsEnabled`, `MaxIteratorResultItems`, etc. so that RPC client can benefit from it.

### PR DESCRIPTION
**Pull request by AliceInHunterland**
neo-project/neo-modules#878

Close #3276

### CODE REVIEW

File: pkg/neorpc/result/version.go [hunk 1]
```diff
@@ -14,11 +14,13 @@ type (
...
-		Nonce     uint32   `json:"nonce"`
-		UserAgent string   `json:"useragent"`
-		Protocol  Protocol `json:"protocol"`
+		TCPPort                uint16   `json:"tcpport"`
+		WSPort                 uint16   `json:"wsport,omitempty"`
+		Nonce                  uint32   `json:"nonce"`
+		UserAgent              string   `json:"useragent"`
+		Protocol               Protocol `json:"protocol"`
+		MaxIteratorResultItems int      `json:"maxiteratorresultitems"`
+		SessionEnabled         bool     `json:"sessionenabled"`
```
Comment (AnnaShaleva) [inline pkg/neorpc/result/version.go:23]: You haven't ported the https://github.com/neo-project/neo-modules/pull/878. It should be a separate `rpc` section:
https://github.com/neo-project/neo-modules/blob/12ccd6810248a7d89670a9a8f72c46a33cb39810/src/RpcServer/RpcServer.Node.cs#L98

File: pkg/neorpc/result/version_test.go [hunk 2]
```diff
@@ -30,6 +30,8 @@ func TestVersion_MarshalUnmarshalJSON(t *testing.T) {
     }`
 	responseFromGoNew := `{
         "nonce": 1677922561,
+		"maxiteratorresultitems":0,
+		"sessionenabled":false,
```
Comment (AnnaShaleva) [inline pkg/neorpc/result/version_test.go:34]: Indent.

**Review by AnnaShaleva — CHANGES_REQUESTED** [review]

File: pkg/neorpc/result/version.go [hunk 3]
```diff
@@ -77,6 +90,15 @@ type (
 // prefixHardfork is a prefix used for hardfork names in C# node.
 const prefixHardfork = "HF_"
 
+// MarshalJSON implements the JSON marshaler interface.
+func (r RPC) MarshalJSON() ([]byte, error) {
+	aux := rpcMarshallerAux{
```
Comment (AliceInHunterland) [inline pkg/neorpc/result/version.go:95]: don't understand why linter isn't happy...

File: pkg/neorpc/result/version.go [hunk 4]
```diff
@@ -48,6 +55,12 @@ type (
 		ValidatorsHistory map[uint32]uint32
 	}
 
+	// rpcMarshallerAux is an auxiliary struct used for RPC JSON marshalling.
+	rpcMarshallerAux struct {
+		MaxIteratorResultItems int  `json:"maxiteratorresultitems"`
+		SessionEnabled         bool `json:"sessionenabled"`
+	}
+
```
Comment (AnnaShaleva) [inline pkg/neorpc/result/version.go:63]: We don't need an auxiliary structure for marshalling. It's straightforward.

File: pkg/neorpc/result/version.go [hunk 5]
```diff
@@ -77,6 +90,11 @@ type (
 // prefixHardfork is a prefix used for hardfork names in C# node.
 const prefixHardfork = "HF_"
 
+// MarshalJSON implements the JSON marshaler interface.
+func (r RPC) MarshalJSON() ([]byte, error) {
+	return json.Marshal(rpcMarshallerAux(r))
+}
+
```
Comment (AnnaShaleva) [inline pkg/neorpc/result/version.go:97]: Not needed.

File: pkg/neorpc/result/version.go [hunk 6]
```diff
@@ -109,6 +127,18 @@ func (p Protocol) MarshalJSON() ([]byte, error) {
 	return json.Marshal(aux)
 }
 
+// UnmarshalJSON implements the JSON unmarshaler interface.
+func (r *RPC) UnmarshalJSON(data []byte) error {
+	var aux rpcMarshallerAux
```
Comment (AnnaShaleva) [inline pkg/neorpc/result/version.go:132]: Not needed.

File: pkg/neorpc/result/version_test.go [hunk 7]
```diff
@@ -42,6 +42,10 @@ func TestVersion_MarshalUnmarshalJSON(t *testing.T) {
             "validatorscount": 7,
             "hardforks": [{"name": "Aspidochelone", "blockheight": 123}, {"name": "Basilisk", "blockheight": 1234}]
         },
+        "rpc": {
+            "maxiteratorresultitems": 0,
+            "sessionenabled": false
+        },
```
Comment (AnnaShaleva) [inline pkg/neorpc/result/version_test.go:48]: Can't be 0, add some meaningful values.

File: pkg/services/rpcsrv/server.go [hunk 8]
```diff
@@ -868,6 +868,9 @@ func (s *Server) getVersion(_ params.Params) (any, *neorpc.Error) {
 		TCPPort:   port,
 		Nonce:     s.coreServer.ID(),
 		UserAgent: s.coreServer.UserAgent,
+		RPC: result.RPC{
+			MaxIteratorResultItems: s.config.MaxIteratorResultItems,
+			SessionEnabled:         s.config.SessionEnabled},
```
Comment (AnnaShaleva) [inline pkg/services/rpcsrv/server.go:873]: Move the last brace to the new line.

File: pkg/neorpc/result/version_test.go (hunk 7 above)
Comment (AnnaShaleva) [reply pkg/neorpc/result/version_test.go:48]: Also need to adjust RPC client test (`pkg/rpcclient/rpc_test.go`). Better with some real response from C# if you're able to build C# node from source.

File: pkg/rpcclient/rpc_test.go [hunk 9]
```diff
@@ -1986,7 +1990,7 @@ func wrapInitResponse(r *params.In, resp string) string {
 	var response string
 	switch r.Method {
 	case "getversion":
-		response = `{"id":1,"jsonrpc":"2.0","result":{"protocol":{"network":42},"tcpport":20332,"wsport":20342,"nonce":2153672787,"useragent":"/NEO-GO:0.73.1-pre-273-ge381358/"}}`
+		response = `{"id":1,"jsonrpc":"2.0","result":{"protocol":{"network":42},"tcpport":20332,"wsport":20342,"nonce":2153672787,"useragent":"/NEO-GO:0.73.1-pre-273-ge381358/", "rpc": {"maxiteratorresultitems": 100, "sessionenabled": false}}}`
```
Comment (AnnaShaleva) [inline pkg/rpcclient/rpc_test.go:1993]: What I meant is that we should adjust the following testcase: https://github.com/nspcc-dev/neo-go/blob/be1b97d04e416d110de63b8f4ddd817dd8363c9d/pkg/rpcclient/rpc_test.go#L1021

But this adjustment is also valid, let's keep it.

Comment (AliceInHunterland) [reply pkg/rpcclient/rpc_test.go:1993]: they seem to be connected

File: pkg/rpcclient/rpc_test.go [hunk 10]
```diff
@@ -1024,7 +1024,7 @@ var rpcClientTestCases = map[string][]rpcClientTestCase{
 			invoke: func(c *Client) (any, error) {
 				return c.GetVersion()
 			},
-			serverResponse: `{"id":1,"jsonrpc":"2.0","result":{"tcpport":20332,"wsport":20342,"nonce":2153672787,"useragent":"/NEO-GO:0.73.1-pre-273-ge381358/"}}`,
+			serverResponse: `{"id":1,"jsonrpc":"2.0","result":{"tcpport":20332,"wsport":20342,"nonce":2153672787,"useragent":"/NEO-GO:0.73.1-pre-273-ge381358/", "protocol": {"network": 2, "hardforks": {}}, "rpc": {"maxiteratorresultitems": 100, "sessionenabled": false}}`,
```
[the remark this answers is not in the document]
Comment (AliceInHunterland) [reply pkg/rpcclient/rpc_test.go:1027]: actually serverResponse is not really useful in this test.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**